temporal-sdk/src/main/java/io/temporal/client/LazyUpdateHandleImpl.java

```diff
@@ -109,4 +113,18 @@ public CompletableFuture<T> getResultAsync(long timeout, TimeUnit unit) {
   public CompletableFuture<T> getResultAsync() {
     return this.getResultAsync(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
   }
+
+  WorkflowClientCallsInterceptor.PollWorkflowUpdateOutput<T> pollUntilComplete(
+      long timeout, TimeUnit unit) {
+    synchronized (this) {
```

Comment:
Using synchronized like this can be really problematic with virtual threads because the virtual thread will be pinned while executing `pollWorkflowUpdate` https://mikemybytes.com/2024/02/28/curiosities-of-java-virtual-threads-pinning-with-synchronized/

Reply:
Ah, interesting didn't realize that.

Reply:
Yeah it is a very unfortunate issue with the current virtual thread limitation. My current stance  is avoid IO in synchronized  blocks and remove any I see